### PR TITLE
Loc Checkin

### PR DESCRIPTION
**Pull request by cristianosuzuki77**
Loc Checkin

### CODE REVIEW

**Comment by cristianosuzuki77** [comment]
@wli3

File: src/Compilers/Core/Portable/xlf/CodeAnalysisResources.de.xlf [hunk 1]
```diff
@@ -571,12 +571,12 @@
...
-        <target state="new">Global analyzer config section name '{0}' is invalid as it is not an absolute path. Section will be ignored. Section was declared in file: '{1}'</target>
+        <target state="translated">Der Name des Konfigurationsabschnitts "{0}" des globalen Analysemoduls ist ungültig, weil es sich nicht um einen absoluten Pfad handelt. Der Abschnitt wird ignoriert. Der Abschnitt wurde in der Datei "{1}" deklariert.</target>
         <note>{0}: invalid section name, {1} path to global config</note>
       </trans-unit>
       <trans-unit id="WRN_InvalidGlobalSectionName_Title">
         <source>Global analyzer config section name is invalid as it is not an absolute path. Section will be ignored.</source>
-        <target state="new">Global analyzer config section name is invalid as it is not an absolute path. Section will be ignored.</target>
+        <target state="translated">Der Name des Konfigurationsabschnitts "{0}" des globalen Analysemoduls ist ungültig, weil es sich nicht um einen absoluten Pfad handelt. Der Abschnitt wird ignoriert.</target>
```
Comment (wli3) [inline src/Compilers/Core/Portable/xlf/CodeAnalysisResources.de.xlf:579]: @cristianosuzuki77  this looks wrong. The original does not have {0}

Comment (wli3) [reply src/Compilers/Core/Portable/xlf/CodeAnalysisResources.de.xlf:579]: How quick could you hand it back? Due to this we cannot merge this change. Build is failing

Comment (cristianosuzuki77) [reply src/Compilers/Core/Portable/xlf/CodeAnalysisResources.de.xlf:579]: Working on it.

Comment (cristianosuzuki77) [reply src/Compilers/Core/Portable/xlf/CodeAnalysisResources.de.xlf]: @wli3 I'm flagging to the translation team. Meanwhile, I reverted the translation until this is resolved. ETA Early Monday

**Comment by wli3** [comment]
@JoeRobich could you help merge this PR?

File: src/Analyzers/Core/Analyzers/xlf/AnalyzersResources.zh-Hans.xlf [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
-  <file datatype="xml" source-language="en" target-language="zh-Hans" original="../AnalyzersResources.resx">
+  <file datatype="xml" source-language="en" target-language="zh-HANS" original="../AnalyzersResources.resx">
```
Comment (JoeRobich) [inline src/Analyzers/Core/Analyzers/xlf/AnalyzersResources.zh-Hans.xlf:3]: Is there a reason to prefer one capitalization over the other? or in the end, does it matter?

Comment (cristianosuzuki77) [reply src/Analyzers/Core/Analyzers/xlf/AnalyzersResources.zh-Hans.xlf:3]: No reason.  Either way should be fine.

**Comment by cristianosuzuki77** [comment]
@wli3 submitted the fix for the German file via "Loc fix" change request.

**Comment by cristianosuzuki77** [comment]
@wli3 Any other feedback for localization?

**Comment by wli3** [comment]
@JoeRobich anything blocking the merge? We are 2 days away from VS loc freeze

**Comment by JoeRobich** [comment]
@wli3 It looks like a unit test may not be using localized string and that is why the Spanish Unit Tests are failing. I'll work on getting a fix for that in today so we can merge this pr.

**Comment by wli3** [comment]
any luck?

File: src/Features/Core/Portable/xlf/FeaturesResources.de.xlf [hunk 3]
```diff
@@ -642,7 +642,7 @@ Stellen Sie sicher, dass Sie den Bezeichner "tt" für Sprachen verwenden, für d
       </trans-unit>
       <trans-unit id="Modifying_source_file_will_prevent_the_debug_session_from_continuing_due_to_internal_error">
         <source>Modifying source file '{0}' will prevent the debug session from continuing due to internal error: {1}</source>
-        <target state="needs-review-translation">Durch das Ändern der Quelldatei "{0}" wird verhindert, dass die Debugsitzung aufgrund eines internen Fehlers fortgesetzt wird: {1}.</target>
+        <target state="translated">Durch das Ändern der Quelldatei "{0}" wird die Fortsetzung der Debugsitzung aufgrund des folgenden internen Fehlers verhindert: {1}.</target>
```
Comment (JoeRobich) [inline src/Features/Core/Portable/xlf/FeaturesResources.de.xlf:645]: Translated string contains `.` not in source string.
```suggestion
        <target state="translated">Durch das Ändern der Quelldatei "{0}" wird die Fortsetzung der Debugsitzung aufgrund des folgenden internen Fehlers verhindert: {1}</target>
```

File: src/Features/Core/Portable/xlf/FeaturesResources.es.xlf [hunk 4]
```diff
@@ -642,7 +642,7 @@ Asegúrese de usar el especificador "tt" para los idiomas para los que es necesa
       </trans-unit>
       <trans-unit id="Modifying_source_file_will_prevent_the_debug_session_from_continuing_due_to_internal_error">
         <source>Modifying source file '{0}' will prevent the debug session from continuing due to internal error: {1}</source>
-        <target state="needs-review-translation">Modificar fuente archivo {0} evitará que la sesión de depuración continua debido a un error interno: {1}</target>
+        <target state="translated">Si se modifica el archivo de código fuente "{0}", se impedirá que continúe la sesión de depuración debido a un error interno: {1}.</target>
```
Comment (JoeRobich) [inline src/Features/Core/Portable/xlf/FeaturesResources.es.xlf:645]: Translated string contains a `.` not present in source string.
```suggestion
        <target state="translated">Si se modifica el archivo de código fuente "{0}", se impedirá que continúe la sesión de depuración debido a un error interno: {1}</target>
```

File: src/Features/Core/Portable/xlf/FeaturesResources.pl.xlf [hunk 5]
```diff
@@ -642,7 +642,7 @@ Pamiętaj, aby nie używać specyfikatora „tt” dla wszystkich języków, w k
       </trans-unit>
       <trans-unit id="Modifying_source_file_will_prevent_the_debug_session_from_continuing_due_to_internal_error">
         <source>Modifying source file '{0}' will prevent the debug session from continuing due to internal error: {1}</source>
-        <target state="needs-review-translation">Zmodyfikowanie pliku źródłowego {0} uniemożliwi kontynuowanie sesji debugowania ze względu na błąd wewnętrzny: {1}.</target>
+        <target state="translated">Zmodyfikowanie pliku źródłowego „{0}” uniemożliwi kontynuowanie sesji debugowania ze względu na błąd wewnętrzny: {1}.</target>
```
Comment (JoeRobich) [inline src/Features/Core/Portable/xlf/FeaturesResources.pl.xlf:645]: Translated string contains `.` not present in source string
```suggestion
        <target state="translated">Zmodyfikowanie pliku źródłowego „{0}” uniemożliwi kontynuowanie sesji debugowania ze względu na błąd wewnętrzny: {1}</target>
```

**Review by JoeRobich — COMMENTED** [review]
Some strings still contain `.` not present in source string

**Comment by JoeRobich** [comment]
@cristianosuzuki77 Could you apply these suggestions and we'll see if we get a passing CI run.

**Comment by cristianosuzuki77** [comment]
For languages certain the translators have confirmed there will be cases where the . will be added, as the grammar requires. 
Some languages will even add extra punctuation, as for Japanese adding ':' when it requires.
Also, the translation team have reviewed all cases where the punctuation was flagged in previous PR across all translations, but I can request them to confirm again over these samples.

**Comment by JoeRobich** [comment]
@cristianosuzuki77 Ah sorry, I didn't have that extra context.

I'll see about a code fix to work with around a possible trailing '.'.

**Comment by Youssef1313** [comment]
@JoeRobich I looked into one of the failing tests:

https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/EditorFeatures/CSharpTest/EditAndContinue/CSharpEditAndContinueAnalyzerTests.cs#L718-L753

I assume the "actual" result differs from the "expected" result because the test is run against some existing version that doesn't yet have the "."?

If that's the case, I think the most straightforward fix is to update the tests to use LocalizableResourceString instead of a normal string, then use `.ToString(CultureInfo.InvariantCulture)`. This will ensure that the test is using the English resource.

**Comment by JoeRobich** [comment]
Tests fixed in https://github.com/dotnet/roslyn/pull/48611

**Comment by JoeRobich** [comment]
> @JoeRobich I looked into one of the failing tests:

@Youssef1313 Thanks for taking a look at the failures. Let me admit that I had to install the Spanish language packs for Windows and VS to finally see what was going on.

**Pull request merged**